### PR TITLE
chore: add adjust android specific config - GRO-500

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **chore**

### Description

Adds android adjust integration following docs here: https://segment.com/docs/connections/destinations/catalog/adjust/#android

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Adjust integration - brian, jon

#### Dev changes

-

</details>

https://artsyproduct.atlassian.net/browse/GRO-500

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -141,8 +143,8 @@ android {
         applicationId "net.artsy.app"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 7
-        versionName "1.0.0"
+        versionCode 189
+        versionName "7.0.0"
```
Comment (brainbicycle) [inline android/app/build.gradle:147]: damn thought I removed this

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Android user-facing changes (Adjust integration - brian, jon)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 544d706e0e0b6ccc52f90583cbe3cda7da4cbf45
</p>

**Review by lordkiz — COMMENTED** [review]
@brainbicycle I am so sorry for getting to this late. I somehow missed it.
I don't think we need to directly add adjust's sdk at all. We just need to add Adjust as a destination on Segment and attribution data will be automatically forwarded. 
This is because we are already using the integration: `@segment/analytics-react-native-adjust`
```javascript
analytics
      .setup(WRITE_KEY,
        {
          using: [Braze, Adjust],
        }
      )
```
We only need to specify that we want attribution data to be tracked. In that case the configuration should look like this:
```javascript
{
    using: [Braze, Adjust],
    trackAttributionData: true,
    // optionally collect device id on android
    android: {
        flushInterval: 60,
        collectDeviceId: true
    },
}
```

Segment will already do all these setup when we initialise:
https://github.com/segmentio/analytics-react-native/blob/master/packages/core/android/src/main/java/com/segment/analytics/reactnative/core/RNAnalytics.kt


Patch onResume
https://github.com/segmentio/analytics-react-native/blob/master/packages/integrations/patches/%40segment/analytics-react-native-adjust/android/src/main/java/com/segment/analytics/reactnative/integration/adjust/IntegrationModule.kt

**Comment by brainbicycle** [comment]
Going to close for now while we wait for feedback on integration, will revisit then.

**Comment by jonallured** [comment]
After some back and forth with our rep at Adjust we are indeed getting attribution data via the cloud-based syncing BUT we are not getting session info to connect with these events. I showed this PR to that rep and he agreed we should make this change to fix this.

More here!

https://artsyproduct.atlassian.net/browse/GRO-500

**Pull request merged**